src/app/clusters/bindings/bindings.cpp

```diff
@@ -120,55 +163,127 @@ bool emberAfBindingClusterBindCallback(app::CommandHandler * commandObj, const a
         }
     }
 
-    emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_SUCCESS);
-    return true;
+    return CHIP_NO_ERROR;
 }
 
-bool emberAfBindingClusterUnbindCallback(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
-                                         const Commands::Unbind::DecodableType & commandData)
-{
-    NodeId nodeId             = commandData.nodeId;
-    GroupId groupId           = commandData.groupId;
-    ClusterId clusterId       = commandData.clusterId;
-    EndpointId remoteEndpoint = commandData.endpointId;
-    EndpointId localEndpoint  = commandPath.mEndpointId;
-    FabricIndex fabricIndex   = commandObj->GetAccessingFabricIndex();
-    EmberBindingTableEntry bindingEntry;
-
-    ChipLogDetail(Zcl, "RX: UnbindCallback");
+} // namespace
 
-    if ((groupId != 0 && nodeId != 0) || (groupId == 0 && nodeId == 0))
+CHIP_ERROR BindingTableAccess::Read(FabricIndex fabricIndex, const ConcreteReadAttributePath & path,
+                                    AttributeValueEncoder & encoder)
+{
+    switch (path.mAttributeId)
     {
-        ChipLogError(Zcl, "Binding: Invalid request");
-        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_MALFORMED_COMMAND);
-        return true;
+    case BindingList::Id:
+        return ReadBindingTable(fabricIndex, path.mEndpointId, encoder);
+    default:
+        break;
     }
-    if (groupId)
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR BindingTableAccess::ReadBindingTable(FabricIndex fabricIndex, EndpointId endpoint, AttributeValueEncoder & encoder)
+{
+    DeviceLayer::AttributeList<Structs::BindingEntry::Type, EMBER_BINDING_TABLE_SIZE> bindingTable;
```

Comment:
Why?  Why can't we just encode as we go?  Why does the whole list need to be built on the stack?  It doesn't.